src/neptune_scale/sync/sync_process.py

```
@@ -595,47 +607,60 @@ def _is_fatal_error(ex: Exception) -> bool:
+        self._aio_transport: Optional[AsyncioRequestsTransport] = None
 
     def work(self) -> None:
         try:
-            while file_upload_requests := self._operations_repository.get_file_upload_requests(10):
+            while file_upload_requests := self._operations_repository.get_file_upload_requests(
+                self._max_concurrent_uploads
+            ):
                 logger.debug(f"Have {len(file_upload_requests)} file upload requests to process")
 
+                # No networking was done yet - initialize lazily
                 if not self._api_client:
                     self._api_client = backend_factory(self._neptune_api_token)
 
+                    self._aio_loop = asyncio.new_event_loop()
+                    asyncio.set_event_loop(self._aio_loop)
+
+                    self._aio_session = requests.Session()
+                    self._aio_transport = AsyncioRequestsTransport(session=self._aio_session, session_owner=False)
+
                 destination_paths = [file.destination for file in file_upload_requests]
                 storage_urls = fetch_file_storage_urls(self._api_client, self._project, destination_paths)
 
-                for file in file_upload_requests:
-                    try:
-                        upload_file(file.source_path, file.mime_type, storage_urls[file.destination])
-                        if file.is_temporary:
-                            logger.debug(f"Removing temporary file {file.source_path}")
-                            Path(file.source_path).unlink(missing_ok=True)
+                assert self._aio_loop  # mypy
+
+                # Fan out file uploads as async tasks, and block until all tasks are done.
+                # Passing `return_exceptions=True` to gather() will ensure that an exception
+                # in a single task will not interrupt waiting for the other tasks: we're basically
+                # silencing any exceptions here.
```

If we're silencing exceptions, then we should probably remove the exception handling logic, right? It's now duplicated.

---

It's still necessary for handling errors in fetching file upload urls and repository operations. self._upload_file() should not raise an exception anyway, so it's just a safeguard in case it starts doing so at some point. I'll figure out how to change the comment to better reflect that